tests/modules/decrosstalk/test_qc_plotting.py

```diff
@@ -23,6 +25,107 @@
 matplotlib.use('Agg')
 
 
+@pytest.fixture
+def decrosstalk_data_dir():
+    this_dir = pathlib.Path(__file__).parent.resolve()
+    data_dir = this_dir / 'data/qc_plotting'
+    return data_dir
+
+
+@pytest.fixture
+def decrosstalk_input_json_name(
+        decrosstalk_data_dir):
+    input_json_name = decrosstalk_data_dir / 'DECROSSTALK_example_input.json'
+    return input_json_name
+
+
+@pytest.fixture
+def decrosstalk_output_json_name(
+        decrosstalk_data_dir):
+    output_json_name = decrosstalk_data_dir / 'DECROSSTALK_example_output.json'
+    return output_json_name
+
+
+@pytest.mark.parametrize(
+        "trace, maxdex, maxval",
+        [(np.array([np.NaN, np.NaN, np.NaN]), -1, None),
+         (np.array([1.0, np.NaN, 1.2, 0.9]), 2, 0.2),
+         (np.array([1.0, 2.2, 1.3, 0.8]), 1, 1.05)])
+def test_get_nanmaxed_timestep(
+        trace, maxdex, maxval):
+    (actual_dex, actual_val) = get_nanmaxed_timestep(trace)
+    assert actual_dex == maxdex
+    if maxval is None:
+        assert actual_val is None
+    else:
+        np.testing.assert_allclose(actual_val, maxval)
+
+
+@pytest.mark.parametrize(
```

Comment:
Should you test the all NaNs case for these code as well?

Reply:
Good catch.